src/directives/element/slot.js

```diff
@@ -38,12 +39,48 @@ export default {
         elseBlock._context = this.vm
         content.appendChild(elseBlock)
       }
-      const scope = host
+      let scope = host
         ? host._scope
         : this._scope
-      this.unlink = context.$compile(
-        content, host, scope, this._frag
-      )
+
+      if (this.params.plugin) {
+        if (!host) {
+          // TODO warn to use plugin only within components
+        }
+        // copy all remaining attributes from slot to all direct children
+        let attrs = this.el.attributes
+        for(let i = attrs.length - 1; i >= 0; i--) {
+          let name = attrs[i].name,
+            value = attrs[i].value,
+            children = content.children
+          for(let j = 0, len = children.length;  j < len; j++) {
+            // TODO warn if child can't have attributes?
+            if (!(children[j].hasAttribute(name) ||
+                (name[0]===':' && children[j].hasAttribute('v-bind'+name)) ||
+                (name[0]==='@' && children[j].hasAttribute('v-on:'+name.slice(1)))
+              )) {
+              children[j].setAttribute(name,value)
+            }
+          }
+        }
+        // use v-for scope when available
+        scope = this._frag ? this._frag.scope : scope || {}
+
+        // add all data from context to scope
+        for(let data in context._data) {
+          if (scope[data] == null) {
+            defineReactive(scope,data,context._data[data])
+          }
+        }
+        // add all data from host to scope with prefixed names
+        for(let data in host._data) {
+          defineReactive(scope,'_'+data,host._data[data])
```

Comment:
best way I can think of to avoid naming conflict of context and host.
Forces the user of `plugin` to prefix the variables in his markup when using slot
````html
<slot plugin :text="_text"></slot>
```
But the functionality outweighs the inconsistency (and it is hidden from the user, as it should be)